src/zkl/spec.rs

```diff
@@ -0,0 +1,55 @@
+pub struct ProofSpec {
+    pub attribute_count: u32,
+    pub disclosed_attributes: Vec<DisclosedAttribute>,
+    pub clauses: Vec<ProofSpecClauseType>
+}
+
+/// Used for crossing ffi boundary like 'C'
+pub struct ProofSpecBuilder {
```

Comment:
The 3 fields are same as struct before except for visibility, is the comment above meant to explain this? I still don't get it.

Reply:
To cross FFI boundaries, you pass a pointer to a mutable class that has convenience methods.